

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="Python2.7" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="Python2.7" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Analysis pipeline steps and reports &mdash; UTAP 1.0.0 documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Installation" href="installation.html" />
    <link rel="prev" title="UTAP demo website" href="demo-site.html" /> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> UTAP
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="demo-site.html">UTAP demo website</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Analysis pipeline steps and reports</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#analysis-pipeline-steps">Analysis pipeline steps</a></li>
<li class="toctree-l2"><a class="reference internal" href="#pipeline-report">Pipeline report</a></li>
<li class="toctree-l2"><a class="reference internal" href="#annotation-file">Annotation file</a></li>
<li class="toctree-l2"><a class="reference internal" href="#examples-of-reports">Examples of reports</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="user-guide.html">User guide</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">UTAP</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Analysis pipeline steps and reports</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/rst/rna-mars-seq-results.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="analysis-pipeline-steps-and-reports">
<h1>Analysis pipeline steps and reports<a class="headerlink" href="#analysis-pipeline-steps-and-reports" title="Permalink to this headline">¶</a></h1>
<div class="section" id="analysis-pipeline-steps">
<h2>Analysis pipeline steps<a class="headerlink" href="#analysis-pipeline-steps" title="Permalink to this headline">¶</a></h2>
<ol class="arabic simple">
<li>Trims adapter sequences</li>
<li>Runs FastQC on the trimmed sequences for quality control of the samples, in parallel with the steps that follow</li>
<li>Maps reads to the selected reference genome</li>
<li>Adds UMI and gene information to the reads</li>
<li>Quantifies gene expression by counting reads</li>
<li>Counts UMI's for cases of PCR bias</li>
<li>Detects Differentially Expressed (DE) genes for a model with a single factor</li>
</ol>
<p>Steps 3 and 5 are performed only for Mars-Seq</p>
<p>Step 6 is performed only if DESeq2 is selected</p>
<img alt="../_images/pipeline.png" src="../_images/pipeline.png" />
</div>
<div class="section" id="pipeline-report">
<h2>Pipeline report<a class="headerlink" href="#pipeline-report" title="Permalink to this headline">¶</a></h2>
<p>Upon completion of the analysis, you will be sent an email with links to the results report.</p>
<p>The report includes several sections:</p>
<ol class="arabic">
<li><p class="first">Sequencing and Mapping QC</p>
<blockquote>
<div><ol class="loweralpha simple">
<li><strong>Figure 1</strong> - Plots the average quality of each base across all reads. Qualities of 30 (predicted error rate 1:1000) and above are good</li>
<li><strong>Figure 2</strong> - Histogram showing the number of reads for each sample in the raw data</li>
<li><strong>Figure 3</strong> - Histogram showing the percentage of reads discarded after trimming the adapters (after removing adapters, short, polyA/T and low quality reads are discarded by the pipeline)</li>
<li><strong>Figure 4</strong> - Histogram with the number of reads for each sample in each step of the pipeline</li>
<li><strong>Figure 5</strong> - Plots sequence coverage on and near gene regions</li>
<li><dl class="first docutils">
<dt><strong>Figure 6</strong> -</dt>
<dd><ol class="first last lowerroman">
<li>Histogram showing the percentage of mapped reads (both uniquely and not uniquely) per sample</li>
<li>Histogram showing the percentage of the uniquely mapped reads that mapped to genes (included genes must have at least 5 reads)</li>
</ol>
</dd>
</dl>
</li>
</ol>
</div></blockquote>
</li>
<li><dl class="first docutils">
<dt>Exploratory Analysis</dt>
<dd><ol class="first last loweralpha simple">
<li><strong>Figure 7</strong> - Heatmap plotting the highly-expressed genes (above 5% of total expression). For example, the expression of gene RN45S in sample SRR3112243 constitutes 15% of the total expression</li>
<li><strong>Figure 8</strong> - Heatmap of Pearson correlation between samples according to gene expression values</li>
<li><strong>Figure 9</strong> - Clustering dendogram of the samples according to gene expression</li>
<li><dl class="first docutils">
<dt><strong>Figure 10</strong> - PCA analysis</dt>
<dd><ol class="first last lowerroman">
<li>Histogram of % explained variability for each PC component</li>
<li>PCA plot of PC1 vs PC2</li>
<li>PCA plot of PC1 vs PC3</li>
</ol>
</dd>
</dl>
</li>
</ol>
</dd>
</dl>
</li>
<li><p class="first">Differential Expression Analysis (this section exists only if you run the DESeq2 analysis) - a table with the number of differentially expressed genes (DE) in each category (up/down) for the different contrasts.  In addition, links for p-value distribution, volcano plots and heatmaps, as well as a table of the DE genes with dot plots of their expression values are also provided</p>
</li>
<li><p class="first">Bioinformatics Pipeline Methods - description of pipeline methods</p>
</li>
<li><p class="first">Links to additional results - links for downloading tables with raw, normalized counts, log normalized values (rld), and statistical data of contrasts. In cases of models with batches, &quot;combat&quot; values were calculated (instead of rld) using the &quot;sva&quot; package, providing batch corrected normalized log2 count values.</p>
</li>
</ol>
</div>
<div class="section" id="annotation-file">
<h2>Annotation file<a class="headerlink" href="#annotation-file" title="Permalink to this headline">¶</a></h2>
<p>For counts of the reads per gene, we use annotation files (gtf format) from &quot;Ensembl&quot; or &quot;GENCODE&quot;. In MARS-seq analysis, we extend the 3' UTR exon away from the transcript on the DNA and extend or cut the 3&quot; UTR exon towards the 5' direction on the mRNA.</p>
</div>
<div class="section" id="examples-of-reports">
<h2>Examples of reports<a class="headerlink" href="#examples-of-reports" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://bip.weizmann.ac.il/rna-seq">RNA-Seq example</a></p>
<p><a class="reference external" href="https://bip.weizmann.ac.il/mars-seq">MARS-Seq example</a></p>
<p>Note: This example analysis demonstrates a good starting point, and not necessarily an end result.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="installation.html" class="btn btn-neutral float-right" title="Installation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="demo-site.html" class="btn btn-neutral" title="UTAP demo website" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Refael Kohen.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.0.0',
            LANGUAGE:'Python2.7',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  <script type="text/javascript" src="../_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>